### ISSUE TITLE
Feature request: Import from JSON?

### ISSUE THREAD

**Issue by arcanemachine** [MichaelMure/git-bug#1065]
Hello, I know you're busy and this amazing project is already loaded with things to do. I am just wondering at the feasability of something I would like to do: import a JSON file into a freshly-cloned and populate a `git-bug` issue tracker with that data (which would be exported by `git bug ls --format json`). Sort of like a bridge, but with JSON instead of a specific git provider.

The background is that I've been having trouble with the bridges, both GitLab (I've tried the dev branch, still having issues), and GitHub (this one I should probably troubleshoot more).

Ultimately, what it comes down to is that I really like the `git-bug` interface, but my issues in dealing with the bridges led me to the realization that `git-bug` allows you to export the data as JSON. Because I use the issue tracker as a personal to-do list + notepad (and there are no other lightweight TUI programs (that I could find) that match the `git-bug` experience), if I was able to take that exported data and import it into a copy of the repo on another machine, then I would be able to use the awesome `git-bug` interface, with full portability, without any dependence on bridges which could break compatibility at any moment.

I hope this idea makes sense. More importantly, I hope it could be implemented without too much effort.

Thanks

**Comment by smoyer64** [comment]
Maybe I'm missing something but I use `git-bug` on several machines and just use `git bug push` and `git bug pull`.  The main problem with this technique is that you need to create the identity on only one machine and copy it to the other machines.  Ultimately, you're also going to want one identity even across repositories too (like you have at GitHub.)  Perhaps a DID?

Except for sharing between machines, I can envision other reasons for what amounts to a JSON bridge!

**Comment by arcanemachine** [comment]
Thanks for the reply. I've just been having issues getting any form of portability working (tried GitLab bridge (even with new dev branch), `git bug push/pull`), some of which seem to be related to #1003. I got it working through a GitHub bridge, but I had to set up a repo just for the issues since the original repo is on GitLab.

I do think it would be nice to have a bridge using a protocol that won't break due to uncontrollable changes from external sources (e.g. API changes on third-party platforms), but I understand that this project doesn't have as many contributers as it deserves.

**Comment by smoyer64** [comment]
I'll have to run some tests but this week is pretty busy.  I'm thinking the process of using `git-bug` on the second machine is probably the critical step and I'll document it if I get it working.

**Comment by arcanemachine** [comment]
Well I got `git bug push/pull` working... Not sure what I did differently this time. I just created a new user, did a `git bug pull`, and everything worked as one would expect. Glad to have multiple backup options working!

I'm still using the `dev` build that I compiled myself FWIW.

**Comment by mcepl** [comment]
> Maybe I'm missing something but I use `git-bug` on several machines and just use `git bug push` and `git bug pull`. The main problem with this technique is that you need to create the identity on only one machine and copy it to the other machines. Ultimately, you're also going to want one identity even across repositories too (like you have at GitHub.) Perhaps a DID?

JSON files are easier to back up, and also it adds robustness for the situations when `git-bug` is broken. Like [right now](https://github.com/MichaelMure/git-bug/issues/719#issuecomment-1998087613).

**Comment by mcepl** [comment]
This is still completely alive.

**Comment by mcepl** [comment]
I hate this GitHub bot, and it seems hugely disrespectful to your users to use it.

**Comment by arcanemachine** [comment]
> I hate this GitHub bot, and it seems hugely disrespectful to your users to use it.

Do you really think insulting the developer is going to encourage them to spend more time on this project?

**Comment by mcepl** [comment]
And do you think that openly ignoring and disrespecting collaborators on your project will encourage them to help you? Of course, there are probably so endless flocks of collaborators eager to contribute, that the owner of the project doesn’t want anybody else to help. I guess.

**Comment by MichaelMure** [comment]
There is no endless flock of collaborator. Unfortunately, there is also a shortage of maintainer.

I'm not a fan of that bot either, but as this is a collaborative project, not everything is done your way. And that's ok

**Comment by mcepl** [comment]
He is not dead yet!

**Comment by sudoforge** [comment]
i thought of a use case i wasn't thinking of earlier: bulk issue creation.

other than importing issues from a third party source (bridge), we don't currently have a way to create issues en masse. i'm not sure i think JSON is the right data format for doing this, but i can totally see the need for a bulk import, and since we export JSON, importing it makes sense.

**Comment by mcepl** [comment]
Just to the issue of the stale bot (and I don’t think we need to elaborate on this further): the only good solution of overloading issue trackers in my opinion is fierce and immediate triage on incoming issue reports. Anything else leads to discarding valuable user experience, or to the death of issue trackers, as described in [my other comment](https://github.com/martanne/vis/issues/1145#issuecomment-1756281015).

**Comment by sudoforge** [comment]
@mcepl, the bot (rather, the pipeline that does this) is staying.

`git-bug` is an open source, volunteer-driven project. "volunteer", in this context, means that people are donating their time (sometimes very limited due to other responsibilities), for free, in order to drive a project forward.

things will slip through the cracks. **the bot helps prevent this.**

you are making statements like this:

> the only good solution of overloading issue trackers in my opinion is fierce and immediate triage on incoming issue reports

while not recognizing the fact that without the ability to dedicate one's full attention on a project, this is not possible, especially so in one with a large community and limited maintainers.

i have stated multiple times, as does the bot itself, that it does not close issues. the video you've linked to clearly focuses on automatic closure of issues being the problem, not automatic labeling and commenting (which notifies the maintainers).

to summarize your involvement in this thread:

- you aren't offering help
- you don't seem to comprehend, or at least have not acknowledged, that maintainers (at least, on this project) are **real people** with jobs, lives, and still _donate their time for free_ in order to push a project they're passionate about forward
- you continue to shift the conversation away from what the original poster needed, and have done so on multiple occasions
- you are positioning himself as an authority without context - as in, "let me tell you how you should run your project"

this is not welcome.
